### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by miken32**
### Changes
Clarify location of common modes and regexps, improve grammar, and use consistent case for "JavaScript" and "highlight.js"

### Checklist
Doesn't apply here because it's just documentation

### CODE REVIEW

File: docs/language-guide.rst [hunk 1]
```diff
@@ -253,10 +253,10 @@ Illegal symbols
 ---------------
 
 Another way to improve language detection is to define illegal symbols for a mode.
-For example in Python first line of class definition (``class MyClass(object):``) cannot contain symbol "{" or a newline.
-Presence of these symbols clearly shows that the language is not Python and the parser can drop this attempt early.
+For example, in Python the first line of a class definition (``class MyClass(object):``) cannot contain the symbol ``{`` or a newline.
+The presence of these symbols clearly shows that the language is not Python, and the parser can drop this attempt early.
 
-Illegal symbols are defined as a a single regular expression:
+Illegal symbols are defined as a single regular expression:
```
Comment (joshgoebel) [inline docs/language-guide.rst:259]:
```suggestion
Illegal symbols are defined using a single regular expression:
```

File: docs/language-guide.rst [hunk 2]
```diff
@@ -269,14 +269,13 @@ Illegal symbols are defined as a a single regular expression:
 Pre-defined modes and regular expressions
 -----------------------------------------
 
-Many languages share common modes and regular expressions. Such expressions are defined in core highlight.js code
-at the end under "Common regexps" and "Common modes" titles. Use them when possible.
+Many languages share common modes and regular expressions. Such expressions are defined in `lib/modes.js`. Once compiled, they be found near the end of the core `highlight.js` code under "Common regexps" and "Common modes" titles. Use them when possible.
```
Comment (joshgoebel) [inline docs/language-guide.rst:272]: Often times the compiler code is entirely unreadable (minified) does it even make sense to send somehere here to reference it?

Comment (joshgoebel) [reply docs/language-guide.rst:272]: Github link instead maybe?

File: docs/language-guide.rst [hunk 3]
```diff
@@ -269,14 +269,13 @@ Illegal symbols are defined as a a single regular expression:
...
 
 
 Regular Expression Features
 ---------------------------
 
-The goal of Highlight.js is to support whatever regex features Javascript itself supports.  You're using real regular expressions, use them responsibly.  That said, due to the design of the parser, there are some caveats.  These are addressed below.
+The goal of highlight.js is to support whatever regex features JavaScript itself supports.  You're using real regular expressions, use them responsibly.  That said, due to the design of the parser, there are some caveats.  These are addressed below.
```
Comment (joshgoebel) [inline docs/language-guide.rst:278]:
```suggestion
The goal of Highlight.js is to support whatever regex features JavaScript itself supports.  You're using real regular expressions, use them responsibly.  That said, due to the design of the parser, there are some caveats.  These are addressed below.
```

Our official name is indeed capitalized.  If the reference is to the library or npm module specifically then sometimes it should be lowercase, but not here.

Comment (joshgoebel) [reply docs/language-guide.rst:278]: If you're going to fix "JavaScript" can you please do it in all the documentation `rst` files... I see 10 or so mentions of it with the incorrect casing.

File: docs/language-guide.rst [hunk 4]
```diff
@@ -5,7 +5,7 @@ Highlighting overview
 ---------------------
 
 Programming language code consists of parts with different rules of parsing: keywords like ``for`` or ``if``
-don't make sense inside strings, strings may contain backslash-escaped symbols like ``\"``
+don't make sense inside strings, strings may contain backslash-escaped symbols like ``\"``,
 and comments usually don't contain anything interesting except the end of the comment.
 
 In highlight.js such parts are called "modes".
```
Comment (joshgoebel) [inline docs/language-guide.rst:11]:
```suggestion
In Highlight.js such parts are called "modes".
```

**Comment by joshgoebel** [comment]
Thanks for this!  Just needs a bit more work.

**Review by joshgoebel — APPROVED** [review]

**Comment by joshgoebel** [comment]
@miken32 Awesome, merged.

**Review by idev1990 — COMMENTED** [review]
docs/language-guide.rst

**Comment by joshgoebel** [comment]
What?

**Pull request merged**